app/src/main/java/io/appium/uiautomator2/server/mjpeg/MjpegScreenshotStream.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.uiautomator2.server.mjpeg;
+
+import android.app.UiAutomation;
+import android.graphics.Bitmap;
+
+import java.lang.Math;
+import java.util.List;
+import java.util.Iterator;
+import java.util.Locale;
+
+import io.appium.uiautomator2.model.internal.CustomUiDevice;
+import io.appium.uiautomator2.server.ServerConfig;
+import io.appium.uiautomator2.utils.ScreenshotHelper;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class MjpegScreenshotStream extends Thread {
+    private static final UiAutomation uia =
+        CustomUiDevice.getInstance().getInstrumentation().getUiAutomation();
+    private final List<MjpegScreenshotClient> clients;
+    private boolean stopped = false;
+
+    MjpegScreenshotStream(List<MjpegScreenshotClient> clients) {
+        this.clients = clients;
+    }
+
+    @Override
+    public void interrupt() {
+        this.stopped = true;
+        super.interrupt();
+    }
+
+    @Override
+    public void run() {
+        while (!stopped) {
+            if (clients.size() == 0) {
+                try {
+                    sleep(500);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+                continue;
+            }
+
+            // how long each loop should take in milliseconds
+            long targetInterval =
+                Math.round((1.0f / ServerConfig.getMjpegServerFramerate()) * 1000.0f);
+            long start = System.currentTimeMillis();
+
+            byte[] screenshotData = getScreenshot();
+            if (screenshotData.length > 0) {
+                MjpegScreenshotClient client;
```

Comment:
client could be defined inside the loop

Reply:
I don't think there is any harm in reassigned the reference. Unless you feel really strongly about this, I'd like to leave it as is.

Reply:
Let the compiler take care about it. Luckily these are smart enough nowadays to not care too much about minor optimisations. Although, from the readability and code structure perspective it is always better to keep all variable definitions inside blocks where these are used and as close as possible to the place they are used

Reply:
OK, no problem, and you're right. I guess I'm showing my age :-P

Reply:
This will be resolved w/ the upcoming update(s)